src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -713,11 +713,23 @@ private static void CheckRestrictedTypeReceiver(BoundExpression expression, Comp
 
             if (!result.Succeeded)
             {
-                // If the arguments had an error reported about them then suppress further error
-                // reporting for overload resolution. 
-
-                if (!analyzedArguments.HasErrors)
+                if (analyzedArguments.HasErrors)
+                {
+                    // Errors for arguments have already been reported, except for unbound lambdas.
```

Comment:
Actually, there is nothing to revise. The `UnboundLambda` already saves (caches) the `BoundLambda` that is the result of `BindForErrorRecovery`, and it is already used in `SemanticModel`. Saving replacing them explicitly in the bound argument list wouldn't make things different in any observable way.